libraries/chain/controller.cpp

```diff
@@ -1555,22 +1554,15 @@ struct controller_impl {
          // pending LIB. If the pending LIB not found on the head branch then fetch_branch returns an empty branch.
          // Otherwise fetch_branch will return from chain_head to root iff chain_head on pending LIB branch.
          auto branch = new_lib_num <= head_num ? fork_db.fetch_branch(head_id, new_lib_id) : fork_db.fetch_branch(new_lib_id, head_id);
-         try {
+         try { try {
             auto should_process = [&](auto& bsp) {
                // Only make irreversible blocks that have been validated. Blocks in the fork database may not be on our current best head
                // and therefore have not been validated.
                // An alternative more complex implementation would be to do a fork switch here and validate all blocks so they can be then made
                // irreversible. Instead, this moves irreversible as much as possible and allows the next maybe_switch_forks call to apply these
                // non-validated blocks. After the maybe_switch_forks call (before next produced block or on next received block), irreversible
                // can then move forward on the then validated blocks.
-               // In irreversible mode, break every ~500ms to allow other tasks (e.g. get_info, SHiP) opportunity to run. There is a post
-               // for every incoming blocks; enough posted tasks to apply all blocks queued to the fork db.
-               if (irreversible_mode()) {
-                  if (!replaying && fc::time_point::now() - start > fc::milliseconds(500))
```

Comment:
This timed `apply_block` is not needed anymore since `should_terminate` now checks the `app().quit()` flag.

Reply:
I thought we wanted to make sure that the main thread was not stuck doing something for a long time, to allow other tasks (e.g. get_info, SHiP) an opportunity to run (as the comment mentions), in addition to interrupt processing.

Reply:
Good point. Probably best to add this back in.

Reply:
https://github.com/AntelopeIO/spring/pull/1086/commits/2c0ca2e74e0ef89e116b1cc705a2a4547f52dc0b